torch/distributions/constraints.py

```diff
@@ -81,7 +105,7 @@ def check(self, value):
         return (value % 1 == 0) & (value >= 0)
 
 
-class _IntegerInterval(Constraint):
+class integer_interval(Constraint):
```

Comment:
In the docstrings, can you elaborate on how the domain is supported over varying dimensions? E.g., with `integer_interval`, it's `[lower_bound, upper_bound]` where `*_bound` are arbitrary ndarrays. With `boolean`, it's restricted to scalar 0 and 1.